models/src/node/crd/v2.rs

```diff
@@ -242,3 +271,201 @@ impl BottlerocketShadowStatus {
+        let new_shadow = BottlerocketShadow {
+            metadata: ObjectMeta {
+                /// The converted object has to maintain the same name, namespace and uid
+                name: previous_metadata.name,
+                namespace: previous_metadata.namespace,
+                uid: previous_metadata.uid,
+                owner_references: previous_metadata.owner_references,
+                ..Default::default()
+            },
+            spec,
+            status,
+        };
+        new_shadow
+    }
+}
+#[cfg(test)]
+mod tests {
+    use super::BottleRocketShadow_v1;
+    use super::BottlerocketShadow;
+    use super::BottlerocketShadowSpec;
+    use super::BottlerocketShadowSpec_v1;
+    use super::BottlerocketShadowState;
+    use super::BottlerocketShadowState_v1;
+    use super::BottlerocketShadowStatus;
+    use super::BottlerocketShadowStatus_v1;
+    use serde_json::json;
+
+    #[test]
+    fn test_state_convert() {
+        let original_target_state = vec![
+            (json!("Idle"), json!("Idle")),
+            (json!("StagedUpdate"), json!("StagedAndPerformedUpdate")),
+            (json!("PerformedUpdate"), json!("StagedAndPerformedUpdate")),
+            (json!("RebootedIntoUpdate"), json!("RebootedIntoUpdate")),
+            (json!("MonitoringUpdate"), json!("MonitoringUpdate")),
+        ];
+
+        for (original, target) in original_target_state.into_iter() {
+            let old_state: BottlerocketShadowState_v1 = serde_json::from_value(original).unwrap();
+            let new_state = BottlerocketShadowState::from(old_state);
+            assert_eq!(serde_json::to_value(new_state).unwrap(), target);
+        }
+    }
+
+    #[test]
+    fn test_spec_convert() {
+        let original_target_spec = vec![
+            (
+                json!({
+                    "state": "Idle",
+                    "state_transition_timestamp": null,
+                    "version": null
+                }),
+                json!({
+                    "state": "Idle",
+                    "state_transition_timestamp": null,
+                    "version": null
+                }),
+            ),
+            (
+                json!({
+                    "state": "RebootedIntoUpdate",
+                    "state_transition_timestamp": "2022-07-09T19:32:38.609610964+00:00",
+                    "version": "1.8.0"
+                }),
+                json!({
+                    "state": "RebootedIntoUpdate",
+                    "state_transition_timestamp": "2022-07-09T19:32:38.609610964+00:00",
+                    "version": "1.8.0"
+                }),
+            ),
+        ];
+
+        for (original, target) in original_target_spec.into_iter() {
+            let old_spec: BottlerocketShadowSpec_v1 = serde_json::from_value(original).unwrap();
+            let new_spec = BottlerocketShadowSpec::from(old_spec);
+            assert_eq!(serde_json::to_value(new_spec).unwrap(), target);
```

Comment:
I am using this because this allows me to create the object from json that I can define a list and compare the target and source inside.  To answer why am I not directly put the object in the list, this is because some of the entry for the struct is private and doesn't allow customized setting value.